tests/pipeline/test_slice.py

```diff
@@ -0,0 +1,304 @@
+    AssetID,
+    check_arrays,
+    OpenPrice,
+    parameter_space,
+)
+from zipline.testing.fixtures import WithTradingEnvironment, ZiplineTestCase
+
+
+class SliceTestCase(WithTradingEnvironment, ZiplineTestCase):
+    sids = ASSET_FINDER_EQUITY_SIDS = Int64Index([1, 2, 3])
+    START_DATE = Timestamp('2015-01-31', tz='UTC')
+    END_DATE = Timestamp('2015-03-01', tz='UTC')
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(SliceTestCase, cls).init_class_fixtures()
+
+        day = cls.trading_schedule.day
+        sids = cls.sids
+
+        cls.dates = dates = date_range(
+            '2015-02-01', '2015-02-28', freq=day, tz='UTC',
+        )
+        cls.start_date = dates[14]
+        cls.end_date = dates[18]
+
+        cls.raw_data = DataFrame(
+            data=arange(len(dates) * len(sids), dtype=float).reshape(
+                len(dates), len(sids),
+            ),
+            index=dates,
+            columns=cls.asset_finder.retrieve_all(sids),
+        )
+
+        close_loader = DataFrameLoader(USEquityPricing.close, cls.raw_data)
+
+        cls.engine = SimplePipelineEngine(
+            {USEquityPricing.close: close_loader}.__getitem__,
+            cls.dates,
+            cls.asset_finder,
+        )
+
+    def test_slice(self):
+        """
+        Test that slices can be created by indexing into a term, and that they
+        have the correct shape when used as inputs.
+        """
+        my_asset_column = 0
+        my_asset = self.asset_finder.retrieve_asset(self.sids[my_asset_column])
+        my_asset_only = (AssetID().eq(my_asset.sid))
+
+        returns = Returns(window_length=2)
+        returns_slice = returns[my_asset]
+
+        class UsesSlicedInput(CustomFactor):
+            window_length = 3
+            inputs = [returns, returns_slice]
+
+            def compute(self, today, assets, out, returns, returns_slice):
+                # Make sure that our slice is the correct shape (i.e. has only
+                # one column) and that it has the same values as the original
+                # returns factor from which it is derived.
+                assert returns_slice.shape == (self.window_length, 1)
+                check_arrays(returns_slice[:, 0], returns[:, my_asset_column])
+
+        columns = {
+            'uses_sliced_input': UsesSlicedInput(),
+            'uses_sliced_input_masked': UsesSlicedInput(mask=my_asset_only),
+        }
+
+        # Assertions about the expected slice data are made in the `compute`
+        # function of our custom factor above.
+        self.engine.run_pipeline(
+            Pipeline(columns=columns), self.start_date, self.end_date,
+        )
+
+    def test_adding_slice_column(self):
+        """
+        Test that slices cannot be added as a pipeline column.
+        """
+        my_asset = self.asset_finder.retrieve_asset(self.sids[0])
+        open_slice = OpenPrice()[my_asset]
+
+        with self.assertRaises(UnsupportedPipelineColumn):
```

Comment:
I might call this `UnsupportedPipelineTerm` rather than column.  I'm not sure how we want to refer to slices.